renderdoc/os/win32/win32_stringio.cpp

```diff
@@ -240,7 +240,21 @@ void CreateParentDirectory(const string &filename)
     if(wfn[i] == L'/')
       wfn[i] = L'\\';
 
-  SHCreateDirectoryExW(NULL, wfn.c_str(), NULL);
+  std::vector<wstring> directoriesToCreate;
+  while(true)
+  {
+    DWORD fileAttributes = GetFileAttributesW(wfn.c_str());
+    if(fileAttributes != INVALID_FILE_ATTRIBUTES)
+      break;
+
+    directoriesToCreate.push_back(wfn);
+    wfn = wfn.substr(0, wfn.find_last_of(L'\\'));
```

Comment:
I think this will infinite loop if the passed filename ends with a `\` or `/` character right? We should strip all trailing `\`s after the normalisation above.

EDIT: Following on from the other comment, if this is a relative path and it ends up with `foo\\bar\\blah` becoming `foo`, the `find_last_of` will return `string::npos`, and the `substr` call will return the whole string back again - leading to an infinite loop. I think you'd need to have the `find_last_of` separately, and break out of the loop after pushing the string if there are no more `\` characters.